### ISSUE TITLE
Whoops! It looks like puppet-lint has encountered an error that it doesn't

### ISSUE THREAD

**Issue by bigsampi** [rodjek/puppet-lint#838]
puppet-lint version: 2.3.6
ruby version: 2.5.1-p57
platform: x86_64-darwin17
file path: site/hadoop/manifests/tez.pp
file contents:
```
class hadoop:tez {

  srequire ::hadoop

  file {
    '/root/tez.tgz':
    ensure => file,
    source => 'puppet:///modules/hadoop/tez.tgz',
    owner  => 'root',
    group  => 'root',
    mode   => '0755',
    onlyif => [ ! -d /usr/local/apache-tomcat-8.0.28],
    notify => Exec['untar-tez'];
  }

  # untar tez
  exec { 'untar-tez':
    command     => 'tar -xf /tez.tgz -d /',
    user        => 'root',
    creates     => '/usr/local/apache-tomcat-8.0.28',
    onlyif      => [ ! -d /usr/local/apache-tomcat-8.0.28],
    refreshonly => true,
    require     => File['/root/tez.tgz'];
  }

  file { "/usr/local/apache-tomcat-8.0.28/webapps/tez-ui/config/configs.env":
    content => template('hadoop/tez_configs.env.erb'),
    mode    => '0644',
    owner   => mapred,
    group   => hadoop,
    require  => Exec['untar-tez'],
  }

  ::monit::templatized::process {
    'tez':
      process       => 'tez',
      uid           => 'root',
      gid           => 'root',
      pidfile       => '/var/run/tez.pid',
      start_program => '/usr/local/apache-tomcat-8.0.28/bin/startup.sh',
      stop_program  => '/usr/local/apache-tomcat-8.0.28/bin/shutdown.sh',
      require       => [ File['/usr/local/apache-tomcat-8.0.28/webapps/tez-ui/config/configs.env'];
  }

  exec {
    'monit restart tez':
      command     => '/usr/bin/monit restart tez',
      refreshonly => true,
      logoutput   => on_failure,
      require     => [ File['/usr/local/apache-tomcat-8.0.28/webapps/tez-ui/config/configs.env'];
  }
}


```
error:
```
NoMethodError: undefined method `next_token' for nil:NilClass
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer/token.rb:196:in `find_token_of'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer/token.rb:142:in `next_token_of'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer/token.rb:189:in `find_token_of'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer/token.rb:142:in `next_token_of'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer/token.rb:191:in `find_token_of'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer/token.rb:142:in `next_token_of'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer/token.rb:189:in `find_token_of'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer/token.rb:142:in `next_token_of'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer/token.rb:189:in `find_token_of'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/lexer/token.rb:142:in `next_token_of'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/data.rb:176:in `block in resource_indexes'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/data.rb:171:in `each'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/data.rb:171:in `resource_indexes'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/checkplugin.rb:81:in `resource_indexes'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/plugins/check_resources/ensure_first_param.rb:8:in `check'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/checkplugin.rb:21:in `run'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/checks.rb:61:in `block in run'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/checks.rb:58:in `each'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/checks.rb:58:in `run'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint.rb:205:in `run'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/bin.rb:66:in `block in run'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/bin.rb:62:in `each'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/bin.rb:62:in `run'
/Users/sampathk/.gem/ruby/2.5.0/gems/puppet-lint-2.3.6/bin/puppet-lint:7:in `<top (required)>'
/Users/sampathk/.gem/ruby/2.5.0/bin/puppet-lint:23:in `load'
/Users/sampathk/.gem/ruby/2.5.0/bin/puppet-lint:23:in `<main>'
```

**Comment by usev6** [comment]
If I'm not mistaken, this has been fixed with 8a9091a998. On HEAD it reports an syntax error:

```
ERROR: Syntax error on line 35
Try running `puppet parser validate <file>`
```

**Comment by rodjek** [comment]
Unhandled exception fixed by #875
